### PR TITLE
Make callback registry to not be global and instead put the callback …

### PR DESCRIPTION
**Pull request by mngo87**
…registry into settings.

### CODE REVIEW

File: callbacks.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	// Message type
+	MessageType string
+	// Message major version
+	MessageVersion string
+}
+
+// CallbackFunction is the function signature for a hedwig callback function
+type CallbackFunction func(context.Context, *Message) error
+
+// NewData finds a function that returns a pointer to struct type that a hedwig message data should conform to
```
Comment (maroux) [inline callbacks.go]: `is a function`?

File: models.go [hunk 2]
```diff
@@ -156,17 +160,15 @@ func (m *Message) validate() error {
...
+	callBackFn, err := m.callbackRegistry.getCallbackFunction(CallbackKey{
 		MessageType:    m.dataType,
 		MessageVersion: m.dataSchemaVersion,
-	}
-
-	callBackInfo, ok := callbackRegistry[key]
-	if !ok {
+	})
+	if err != nil {
 		return errors.New("Callback is not defined for message")
```
Comment (maroux) [inline models.go]: return err?

File: settings.go [hunk 3]
```diff
@@ -10,19 +10,12 @@ package hedwig
 import (
 	"context"
```
Comment (maroux) [inline settings.go]: remove empty line

File: callbacks.go (hunk 1 above)
Comment (mngo87) [reply callbacks.go]: heh copy pasta

File: models.go (hunk 2 above)
Comment (mngo87) [reply models.go]: ah yes good point

**Pull request merged**